### DISCUSSION TITLE
High level API with array properties

### DISCUSSION THREAD

**Discussion by Pragmateek** [G-Research/ParquetSharp#421] [Q&A] [answered]
Hi all,
We are currently using the Column API as until now our objects had only primitive types properties.
But we'd like to support simple array properties like `int[]` or `string[]`.
Would it be possible to continue leveraging the Column API or do we have to switch to the Schema like in this more complex example: https://github.com/G-Research/ParquetSharp/blob/master/docs/Nested.md ?
Maybe retrieving the schema generated by the column API and complete it for the array properties.
Or maybe there is a tool that would automatically generate the schema from the .NET type.
Thanks for any help

**Comment by adamreeve** [comment, marked as answer]
Hi @Pragmateek, you can use the `Column` API with array values. For example, this works:
```c#
var columns = new Column[]
{
    new Column<int[]>("int_arrays"),
    new Column<string[]>("string_arrays"),
};

using var writer = new ParquetFileWriter(outputPath, columns);
using var rowGroupWriter = writer.AppendRowGroup();

using var intArrayWriter = rowGroupWriter.NextColumn().LogicalWriter<int[]>();
intArrayWriter.WriteBatch(new int[][]
{
    new [] {0, 1, 2},
    new [] {3, 4},
    new [] {5, 6, 7, 8},
});

using var stringArrayWriter = rowGroupWriter.NextColumn().LogicalWriter<string[]>();
stringArrayWriter.WriteBatch(new string[][]
{
    new [] {"abc", "def"},
    new [] {"ghi"},
    new [] {"jkl", "mno"},
});

writer.Close();
```

**Reply by Pragmateek** [reply]
Hi @adamreeve
Thanks a lot, it is working this way. :)
